GitExtUtils/GitArgumentBuilderExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitCommands
+{
+    public static class GitArgumentBuilderExtensions
+    {
+        private const byte BASE_LENGTH = 4; // 'git '
```

Comment:
I haven't digged into how git is called exactly.
Is this really necessary? I.e., doesn't "git" be part of the executable to be started?
Isn't the full path to `git.exe` prepended if specified in the settings? Does the limit apply to arguments string or to the complete command line? It depends on how `CreateProcess` is called by .NET.

Reply:
This says if we specify application name, then commandline can exclude executable module part. I'll check on how the process is called.

https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw#parameters

Reply:
C# implementation of `System.Diagnostic.Process` always combine application name in command line argument, and always add quotation + ' ' characters, so I modified the code to handle all that.
Reference source code: https://referencesource.microsoft.com/#system/services/monitoring/system/diagnosticts/Process.cs,1952